hamilton/data_quality/pydantic_validators.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any, Type
+
+from pydantic import BaseModel, TypeAdapter, ValidationError
+
+from hamilton.data_quality import base
+from hamilton.htypes import custom_subclass_check
+
+
+class PydanticModelValidator(base.BaseDefaultValidator):
+    """Pydantic model compatibility validator
+
+    Note that this validator uses pydantic's strict mode, which does not allow for
```

Comment:
A code example would be really great here -- especially if you can add to the docs. Ping us if you want help getting that set up, or it's too much of a pain -- you can add a code example and we can add it to the docs.